torch/lib/THCS/generic/THCSTensor.c

```diff
@@ -51,29 +62,38 @@ static void THCSTensor_(rawInit)(THCState *state, THCSTensor *self)
   self->size = NULL;
   self->indices = NULL;
   self->values = NULL;
-  self->nDimension = 0;
+  self->nDimensionI = 0;
+  self->nDimensionV = 0;
   self->contiguous = 0;
   self->nnz = 0;
   // self->flag = TH_TENSOR_REFCOUNTED;
+  self->refcount = 1;
 }
 
-static void THCSTensor_(rawResize)(THCState *state, THCSTensor *self, int nDim, long *size) {
+static void THCSTensor_(rawResize)(THCState *state, THCSTensor *self, int nDimI, int nDimV, long *size) {
   // Only resize valid sizes into tensor.
-  self->size = THRealloc(self->size, sizeof(long)*nDim);
+  self->size = THRealloc(self->size, sizeof(long)*(nDimI + nDimV));
 
-  long d, nDim_ = 0;
-  for (d = 0; d < nDim; d++)
-    if (size[d] > 0)
-      self->size[nDim_++] = size[d];
-  self->nDimension = nDim_;
+  long d, nDimI_ = 0, nDimV_ = 0;
+  for (d = 0; d < nDimI; d++) {
+    if (size[d] > 0) {
+      self->size[nDimI_++] = size[d];
```

Comment:
I didn't get that. It seems to me that if someone constructs a tensor using some non-empty values and indices and passes in size of e.g. (2, 4, 0) or (2, -1), then this might break.